src/ServiceControl/CompositeViews/Messages/MessageViewQueryExtensions.cs

```diff
@@ -0,0 +1,134 @@
+namespace ServiceControl.CompositeViews.Messages
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.IO;
+    using System.Linq;
+    using System.Net.Http;
+    using System.Text;
+    using System.Threading.Tasks;
+    using Nancy;
+    using Newtonsoft.Json;
+    using NServiceBus.Logging;
+    using ServiceBus.Management.Infrastructure.Extensions;
+    using ServiceBus.Management.Infrastructure.Nancy;
+    using ServiceBus.Management.Infrastructure.Nancy.Modules;
+    using HttpStatusCode = System.Net.HttpStatusCode;
+
+    public static class MessageViewQueryExtensions
+    {
+        static JsonSerializer jsonSerializer = JsonSerializer.Create(JsonNetSerializer.CreateDefault());
+        static ILog logger = LogManager.GetLogger(typeof(MessageViewQueryExtensions));
+
+        private static QueryResult EmptyResult = new QueryResult(new List<MessagesView>(), new QueryStatsInfo(string.Empty, DateTime.MinValue, 0, 0));
+
+        public static async Task<dynamic> CombineWithRemoteResults(this BaseModule module, QueryResult localQueryResult)
+        {
+            var httpClientFactory = module.HttpClientFactory; // for testing purposes
+            var remotes = module.Settings.RemoteInstances;
+            var currentRequest = module.Request;
+            var negotiator = module.Negotiate;
+
+            if (remotes.Length > 0)
+            {
+                await UpdateLocalQueryResultWithRemoteData(currentRequest, httpClientFactory, remotes, localQueryResult).ConfigureAwait(false);
+
+                localQueryResult.Messages.Sort(MessageViewComparer.FromRequest(currentRequest));
+            }
+
+            return negotiator.WithPartialQueryResult(localQueryResult, currentRequest);
+        }
+
+        static async Task UpdateLocalQueryResultWithRemoteData(Request currentRequest, Func<HttpClient> httpClientFactory, string[] remotes, QueryResult localQueryResult)
+        {
+            var tasks = new List<Task<QueryResult>>(remotes.Length);
+            // ReSharper disable once LoopCanBeConvertedToQuery
+            foreach (var remote in remotes)
+            {
+                tasks.Add(FetchAndParse(currentRequest, httpClientFactory, remote));
+            }
+
+            var highestTotalCount = localQueryResult.QueryStats.TotalCount;
+            var totalCount = localQueryResult.QueryStats.TotalCount;
+            var lastModified = DateTime.MinValue;
```

Comment:
I think this should also come from the local results. If the local query results have been updated most recently then we want to use that value.